### PR TITLE
Improve default configuration template

### LINKED ISSUE #895: Improve default configuration template
The configuration template does not contain every option as commented lines (e.g permission endpoint or permissions principals per resource)

I propose that we take some of what we've done here:
https://github.com/mozilla-services/kinto-dist/blob/3d17efb2c370822e3db1cc4b5a932684c5e42a4e/config/kinto.ini

And also add a task in the pull-request template to mention that the configuration template should be updated if a new setting was introduced.

### PR DESCRIPTION
**Pull request by jkieberk**
Attempts to fix #895 
NOTE: Throws Test Errors

- added commented out configuration settings (from here):
https://kinto.readthedocs.io/en/latest/configuration/settings.html

- rearranged layout out of file

Fixes #

- [ ] Add documentation.
- [ ] Add tests.
- [ ] Add a changelog entry.
- [ ] Add your name in the contributors file.
- [ ] If you changed the HTTP API, update the [API_VERSION](https://github.com/Kinto/kinto/blob/master/kinto/__init__.py#L15) constant and add an API changelog entry [in the docs](https://github.com/Kinto/kinto/blob/master/docs/api/index.rst)

r? @

### CODE REVIEW

File: kinto/config/kinto.tpl [hunk 1]
```diff
@@ -1,54 +1,155 @@
...
 kinto.storage_url = {storage_url}
+# kinto.storage_max_fetch_size = 10000
+# kinto.storage_pool_size = 25
+# kinto.storage_max_overflow = 5
+# kinto.storage_pool_recycle = -1
+# kinto.storage_pool_timeout = 30
+# kinto.storage_max_backlog = -1
 kinto.cache_backend = {cache_backend}
 kinto.cache_url = {cache_url}
+# kinto.cache_prefix = {cahce_prefix}
```
Comment (jkieberk) [inline kinto/config/kinto.tpl]: This throws a KeyError: u'cache_prefix' even though it is commented out?

**Comment by jkieberk** [comment]
Line 42 throws a KeyError: u'cache_prefix' even though it is commented out? 

I'm able to test with 0 errors with the original file.

**Comment by glasserc** [comment]
Yeah, that's because we "render" the template using the `str.format` method, which doesn't understand comments. You can see that happening at https://github.com/Kinto/kinto/blob/53ddeabd0e7ca68921f5360fd80e2ec3d7ac233b/kinto/config/__init__.py#L24. I don't think you need to make `cache_prefix` be a variable in this way -- you can just put a dummy example value.

File: kinto/config/kinto.tpl [hunk 2]
```diff
@@ -1,54 +1,155 @@
...
+#
+# https://kinto.readthedocs.io/en/latest/configuration/settings.html#scheme-host-and-port
+#
+# kinto.http_host = none
+# kinto.http_scheme = none
+#
 
 #
-# Auth configuration.
+# Logging, Monitoring
```
Comment (Natim) [inline kinto/config/kinto.tpl]: Please put the logging section at the very end of the file

File: kinto/config/kinto.tpl [hunk 3]
```diff
@@ -1,54 +1,155 @@
...
+# optional:
+# kinto.statsd_prefix = kinto-prod
+#
+#
+# Monitoring with New Relic
+#
+# https://kinto.readthedocs.io/en/latest/configuration/settings.html#monitoring-with-new-relic
+#
+# kinto.newrelic_config = none
+# kinto.newrelic_env = none
```
Comment (leplatrem) [inline kinto/config/kinto.tpl]: I don't think those lines are necessary

File: kinto/config/kinto.tpl [hunk 4]
```diff
@@ -59,6 +160,60 @@ kinto.includes = kinto.plugins.default_bucket
...
+
+#
+# Notifications
+#
+# https://kinto.readthedocs.io/en/latest/configuration/settings.html#notifications
+#
+# 
+# Filtering
+#
+# https://kinto.readthedocs.io/en/latest/configuration/settings.html#filtering
```
Comment (leplatrem) [inline kinto/config/kinto.tpl]: This filtering section may be a bit confusing. A very simple example of notification can be useful though (see maybe https://github.com/Kinto/kinto-changes/blob/master/README.rst)

File: kinto/config/kinto.tpl [hunk 5]
```diff
@@ -59,6 +160,60 @@ kinto.includes = kinto.plugins.default_bucket
...
+#
+# kinto.flush_endpoint_enabled = true
+
+#
+# Activating the Permissions Endpoint
+#
+# https://kinto.readthedocs.io/en/latest/configuration/settings.html#activating-the-permissions-endpoint
+#
+# kinto.experimental_permissions_endpoint = true
+
```
Comment (leplatrem) [inline kinto/config/kinto.tpl]: I think those go to the features section

File: kinto/config/kinto.tpl [hunk 6]
```diff
@@ -82,22 +237,43 @@ kinto.includes = kinto.plugins.default_bucket
...
 # https://kinto.readthedocs.io/en/latest/configuration/production.html
 #
-# kinto.statsd_backend = kinto.core.statsd
 # kinto.statsd_url = udp://localhost:8125
 # kinto.statsd_prefix = kinto-prod
-
+#
 # kinto.http_scheme = https
 # kinto.http_host = kinto.services.mozilla.com
-
```
Comment (leplatrem) [inline kinto/config/kinto.tpl]: redundant ?

File: kinto/config/kinto.tpl [hunk 7]
```diff
@@ -82,22 +237,43 @@ kinto.includes = kinto.plugins.default_bucket
...
+#
+# Application Profiling
+#
+# https://kinto.readthedocs.io/en/latest/configuration/settings.html#application-profiling
+#
+# kinto.profiler_enabled = true
+# kinto.profiler_dir = /tmp/profiling
+
+#
 # Production settings
```
Comment (leplatrem) [inline kinto/config/kinto.tpl]: production settings should go relatively on top, since they are more useful than profiling etc.

**Comment by leplatrem** [comment]
Thanks for starting this! There are still some things to improve, but at least it is very good to have an exhaustive list of possible settings!

**Comment by leplatrem** [comment]
Closing for lack of activity :| Feel free to re-open if you want to continue working on it! Thanks again!

**Pull request closed without merging**